components/tikv_kv/src/stats.rs

```diff
@@ -190,9 +191,98 @@ pub struct Statistics {
     // Note that a value comes from either write cf (due to it's a short value) or default cf, we
     // can't embed this `processed_size` field into `CfStatistics`.
     pub processed_size: usize,
+
+    // When getting data from default cf, we can check write cf statistics to decide which method
+    // should be used to get the data.
+    load_data_hint: LoadDataHintStatistics,
+}
+
+#[derive(Default, Debug)]
+struct LoadDataHintStatistics {
+    last_write_next: usize,
+    last_write_next_tombstone: usize,
+    last_write_seek: usize,
+    last_write_seek_tombstone: usize,
+
+    last_write_prev: usize,
+    last_write_prev_tombstone: usize,
+    last_write_seek_for_prev: usize,
+    last_write_seek_for_prev_tombstone: usize,
+
+    last_hint: LoadDataHint,
+}
+
+#[derive(Default, PartialEq, Debug, Clone)]
+pub enum LoadDataHint {
+    #[default]
+    NearSeek,
+    Seek,
+    Get,
 }
 
 impl Statistics {
+    pub fn load_data_hint(&mut self) -> LoadDataHint {
+        let stats = &mut self.load_data_hint;
+        let next_overhead = self.write.next - stats.last_write_next + self.write.next_tombstone
+            - stats.last_write_next_tombstone;
+
+        let hint = if self.write.seek != stats.last_write_seek
+            || next_overhead > SEEK_BOUND as usize
+        {
+            if self.write.seek_tombstone - stats.last_write_seek_tombstone > SEEK_BOUND as usize {
+                // if tombstones are too many, we should use get to get rid of them.
+                LoadDataHint::Get
+            } else {
+                // otherwise, we can use seek to get the data. Seek is faster than get for
+                // continuous values.
+                LoadDataHint::Seek
+            }
```

Comment:
Could you explain more about why the tombstones could be got rid of using `Get` and in what situations the `Seek` is faster, I don't quite get it. Or is it possible to add such a test case?

Reply:
Oh, I was wrong. Seems the default cf would never encounter seek tombstone because the default cf value is deterministic. The first inner value with `seek` must not be tombstone. Though there will be some next tombstone. 

For a single value, get is surely faster than seek. But if there are continuous values, seek + next is better than get overall, see the result
<img width="730" alt="截屏2023-12-28 11 02 59" src="https://github.com/tikv/tikv/assets/13497871/4dfa8310-4e7e-4766-9d5c-8e43ae0e27a5">
<img width="740" alt="截屏2023-12-28 11 02 51" src="https://github.com/tikv/tikv/assets/13497871/8158b044-b4bb-4537-9fc6-483996cfd953">

Reply:
Okay I understand, it seems fine overall. 

Regarding using write cf statistics to infer the situation of the default cf, could it be inaccurate in some rare cases? For instance, if the write cf has quite a few tombstones but they are all short values, and the default cf is continuous but doesn't have too many tombstones, would near seek be more appropriate in this situation?

Reply:
Indeed, I changed the condition that avoid near seek only when over seek bound is changed which indicates near seek fails to find the kv by next. By this, short values of write cf wouldn't make it skip near seek. PTAL again